File: services/networkx_service/src/process/instance/instance_repository.py
import json
from .instance import Instance
from .utils import Utils


class InstanceRepository:
    @staticmethod
    def get_instance(instance_path):
        with open(instance_path) as f:
            data = json.load(f)
            data_json = json.loads(data)

            graph = Utils.construct_graph(data_json['graph'])
            parameters = data_json['parameters']

            return Instance(graph, parameters)

File: services/jgrapht_service/src/process/src/main/java/process/instance/InstanceRepository.java
package process.instance;

import com.jsoniter.JsonIterator;
import com.jsoniter.any.Any;
import org.jgrapht.Graph;
import org.jgrapht.graph.DefaultEdge;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class InstanceRepository {
    public static Instance getInstance(String instancePath) {
        try {
            String jsonString = JsonIterator.deserialize(Files.readString(Paths.get(instancePath))).toString();
            Any rootJSON = JsonIterator.deserialize(jsonString);
            Any graphJSON = rootJSON.get("graph");
            Any parametersJSON = rootJSON.get("parameters");

            Graph<Integer, DefaultEdge> graph = Utils.constructGraph(graphJSON);

            return new Instance(graph, parametersJSON);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}
